### DISCUSSION TITLE
Fail to connect port 8080

### DISCUSSION THREAD

**Discussion by Fisher-Wang** [selkies-project/docker-nvidia-glx-desktop#60] [Q&A] [answered]
### My environment
- Ubuntu 22.04.4 LTS
- Nvidia Driver version: 560.35.03
- Nvidia Container version:
    ```
    libnvidia-container-dev/unknown 1.16.1-1 amd64
    libnvidia-container-tools/unknown,now 1.16.1-1 amd64 [installed,automatic]
    libnvidia-container1-dbg/unknown 1.16.1-1 amd64
    libnvidia-container1/unknown,now 1.16.1-1 amd64 [installed,automatic]
    nvidia-container-runtime/unknown 3.14.0-1 all
    nvidia-container-toolkit-base/unknown,now 1.16.1-1 amd64 [installed,automatic]
    nvidia-container-toolkit/unknown,now 1.16.1-1 amd64 [installed]
    ```

### How to reproduce
1. 
     ```
     git clone https://github.com/selkies-project/docker-nvidia-glx-desktop
     cd docker-nvidia-glx-desktop
     ```

2. uncomment `network_mode: host` in `docker-compose.yml`
3. `sudo docker compose up -d`
4. `nc -zv localhost 8080`

    Then I got the error `nc: connect to localhost (127.0.0.1) port 8080 (tcp) failed: Connection refused`

5. Open a brower and visit http://127.0.0.1:8080

   Then I got this
   ![image](https://github.com/user-attachments/assets/f5b4a5b5-1d30-4e99-8289-7e8f2cbf58f6)
   
Could you help me run it correctly? Thank you very much

**Comment by ehfd** [comment]
Need logs inside the container located `cat /tmp/*.log`

**Comment by Fisher-Wang** [comment]
Hi, here is all my log files at `/tmp/*.log`

[/tmp/dbus.log](https://pastebin.ubuntu.com/p/xfyZJkGw6Q/)
[/tmp/entrypoint.log](https://pastebin.ubuntu.com/p/PhfvwDnn6R/)
[/tmp/selkies-gstreamer-entrypoint.log](https://pastebin.ubuntu.com/p/y4Rt4KdV4h/)
[/tmpsupervisord.log](https://pastebin.ubuntu.com/p/KgyXv7r6gV/)
/tmp/kasmvnc-entrypoint.log: empty
/tmp/nginx.log: empty
/tmp/pipewire.log: empty
/tmp/pipewire-pulse.log: empty
/tmp/wireplumber.log: empty

**Comment by PMohanJ** [comment]
As per  logs, it seems like Xserver if failing. By default the DISPLAY is set to `:20` and there's comment in compose file saying `running on host network with :20 might lead to error` https://github.com/selkies-project/docker-nvidia-glx-desktop/blob/main/docker-compose.yml#L28.
Could set the env var DISPLAY to a random value, for instance `DISPLAY=77` and try again?

**Comment by Fisher-Wang** [comment]
I add `- DISPLAY=:77` to the `environment` entry of `docker-compose.yml`, and restart the docker. Unfortunately it gives the same logs as above.

**Comment by Fisher-Wang** [comment]
And here is the content of `/home/ubuntu/.local/share/xorg/Xorg.77.log`: https://pastebin.ubuntu.com/p/4wZ9XGQkDy/, which may have some useful information

**Comment by PMohanJ** [comment]
I'm not sure of those Xorg logs. @ehfd might look at those when he gets time.

**Comment by ehfd** [comment, marked as answer]
This seems like a textbook case of having an X server already running on that GPU in the host.

Answer: https://github.com/selkies-project/docker-nvidia-glx-desktop/issues/11
